### DISCUSSION TITLE
How to export data form Milvus 2.x

### DISCUSSION THREAD

**Discussion by gobigrassland** [milvus-io/milvus#17088] [Q&A and General discussion]
I have tired to export inserted data accroding to the document from https://github.com/milvus-io/milvus-tools, but i failed.  I read read_milvus_meta.py,  the current version supported two database: mysql & sqlite.  I installed Milvus 2.x standalone with Docker Compose.   
(1) Assume that Milvus 2.x chose sqlite, there should existe meta.sqlite. But i did not find this file.
(2) Assume that Milvus 2.x chose myql, i did not remember set myql' database

**Comment by yhmo** [comment]
This tool is "to migrate data from Milvus(0.10.x or 1.x) to Milvus2.x". It **can't** export data from v2.x.
The data management of v2.x is more complicate than v1.x, not easy to export. We plan to provide interface to subscribe data from v2.x, this function is still waiting for design.

**Reply by xiaofan-luan** [reply]
We will be working on data export on Milvus 2.2~ but so far there seems to have nothing to do

**Reply by gobigrassland** [reply]
Looking forward to your development

**Reply by softhuafei** [reply]
@xiaofan-luan hello, is there any other way to export data from Milvus 2.x to csv or HDF5?

**Reply by xiaofan-luan** [reply]
No. We will support iterator features in 2.3. Otherwise you can use spark to read from S3 and do it your self.
